adapter/internal/oasparser/model/async_api.go

```diff
@@ -0,0 +1,243 @@
+			continue
+		}
+
+		if found {
+			pubOrSubVendorExtensions["x-wso2-disable-security"] = val
+		}
+
+		if channelItem.XAuthType == "None" || channelItem.XWso2DisableSecurity {
+			pubOrSubVendorExtensions["x-wso2-disable-security"] = true
+		}
+
+		security := getSecurityArray(pubOrSubVendorExtensions)
+		operation := NewOperation("GET", security, pubOrSubVendorExtensions)
+
+		var methodsArray []*Operation
+		methodsArray = append(methodsArray, operation)
+
+		// we ignore other topic vendor extensions except x-auth-type and x-wso2-disable-security
+		channelVendorExtensions := map[string]interface{}{}
+		if found {
+			channelVendorExtensions["x-wso2-disable-security"] = val
+		}
+		resource := setOperationSwagger(channel, methodsArray, channelVendorExtensions)
+		logger.LoggerAPI.Info("Resource: ", resource.methods, resource.path, resource.vendorExtensions)
+		resources = append(resources, &resource)
+	}
+
+	return SortResources(resources)
+}
+
+func getSecurityArray(vendorExtensions map[string]interface{}) (security []map[string][]string) {
+	if vendorExtensions["x-scopes"] != nil {
+		securityItem := make(map[string][]string)
+		rawScopes := vendorExtensions["x-scopes"].([]interface{})
+		var scopes []string
+
+		for _, rawScope := range rawScopes {
+			scopes = append(scopes, rawScope.(string))
+		}
+		securityItem["oauth2"] = scopes
+		security = append(security, securityItem)
+	}
+	return security
+}
+
+func (asyncAPI *AsyncAPI) unmarshallAPILevelVendorExtensions(b []byte) error {
+	var apiDef map[string]interface{}
+	if err := json.Unmarshal(b, &apiDef); err != nil {
+		return errors.New("Error while unmarshalling API level vendor extensions." + err.Error())
+	}
+	for key, value := range apiDef {
+		keyl := strings.ToLower(key)
+		if strings.HasPrefix(keyl, "x-") {
+			if asyncAPI.VendorExtensions == nil {
+				asyncAPI.VendorExtensions = map[string]interface{}{}
+			}
+			asyncAPI.VendorExtensions[key] = value
+		}
+	}
+	return nil
+}
+
+// convertExtensibletoReadableFormat unmarshalls the vendor extensible in asyncapi.
+func convertExtensibletoReadableFormatfromAsyncAPI(asyncAPI AsyncAPI) map[string]interface{} {
+	jsnRawExtensible := asyncAPI.VendorExtensions
+	b, err := json.Marshal(jsnRawExtensible)
+	if err != nil {
+		logger.LoggerOasparser.Error("Error marshalling vendor extensions: ", err)
+	}
+
+	var extensible map[string]interface{}
+	err = json.Unmarshal(b, &extensible)
+	if err != nil {
+		logger.LoggerOasparser.Error("Error unmarshalling vendor extensions: ", err)
+	}
+	return extensible
+}
+
+// This method check if the x-wso2-disable-security vendor extension present in the given
+// asyncapi vendor extensions.
+// If found, it will return two bool values which are the following in order.
+// 1st bool represnt the value of the vendor extension.
+// 2nd bool represent if the vendor extension present.
+func resolveAsyncAPIDisableSecurity(asyncAPI AsyncAPI) (bool, bool) {
```

Comment:
This only checks API level `x-wso2-disable-security` right? Shall we rename this to `(asyncAPI *AsyncAPI) isAPILevelSecurityDisabled() bool